test/WitthdrawWorkflow.t.sol

```diff
@@ -0,0 +1,204 @@
+    function setUp() public {
+        vm.deal(_owner, 100 ether);
+        token = new ERC20Mock("Token", "TNK", 18);
+
+        entryPoint = IKintoEntryPoint(address(new EntryPoint{salt: 0}()));
+
+        IAccessRegistry accessRegistryImpl = new AccessRegistry();
+        UUPSProxy accessRegistryProxy = new UUPSProxy{salt: 0}(address(accessRegistryImpl), "");
+
+        accessRegistry = AccessRegistry(address(accessRegistryProxy));
+        IAccessPoint accessPointImpl = new AccessPoint(entryPoint, accessRegistry);
+
+        accessRegistry.initialize(accessPointImpl);
+        accessPoint = accessRegistry.deployFor(address(_user));
+
+        withdrawWorkflow = new WithdrawWorkflow();
+
+        entryPoint.setWalletFactory(address(accessRegistry));
+        accessRegistry.allowWorkflow(address(withdrawWorkflow));
+
+        deployPaymaster(_owner);
+    }
+
+    function testWithdrawERC20ViaPaymaster() public {
+        token.mint(address(accessPoint), defaultAmount);
+
+        UserOperation[] memory userOps = new UserOperation[](1);
+        userOps[0] = createUserOperationWithPaymaster(
+            1,
+            address(accessPoint),
+            address(withdrawWorkflow),
+            accessPoint.getNonce(),
+            _userPk,
+            abi.encodeWithSelector(WithdrawWorkflow.withdrawERC20.selector, IERC20(token), defaultAmount)
+        );
+
+        entryPoint.handleOps(userOps, payable(_user));
+
+        assertEq(token.balanceOf(_user), defaultAmount);
+    }
+
+    function testWithdrawERC20() public {
+        token.mint(address(accessPoint), defaultAmount);
+
+        bytes memory data =
+            abi.encodeWithSelector(WithdrawWorkflow.withdrawERC20.selector, IERC20(token), defaultAmount);
+
+        vm.prank(_user);
+        accessPoint.execute(address(withdrawWorkflow), data);
+
+        assertEq(token.balanceOf(_user), defaultAmount);
+    }
+
+    function testWithdrawNativeViaPaymaster() public {
+        vm.deal(address(accessPoint), defaultAmount);
+
+        abi.encodeWithSelector(WithdrawWorkflow.withdrawNative.selector, defaultAmount);
+
+        UserOperation[] memory userOps = new UserOperation[](1);
+        userOps[0] = createUserOperationWithPaymaster(
+            1,
+            address(accessPoint),
+            address(withdrawWorkflow),
+            accessPoint.getNonce(),
+            _userPk,
+            abi.encodeWithSelector(WithdrawWorkflow.withdrawNative.selector, defaultAmount)
+        );
+
+        entryPoint.handleOps(userOps, payable(address(accessPoint)));
+
+        assertEq(_user.balance, defaultAmount);
+    }
+
+    function testWithdrawNative() public {
+        vm.deal(address(accessPoint), defaultAmount);
+
+        bytes memory data = abi.encodeWithSelector(WithdrawWorkflow.withdrawNative.selector, defaultAmount);
+
+        vm.prank(_user);
+        accessPoint.execute(address(withdrawWorkflow), data);
+
+        assertEq(_user.balance, defaultAmount);
+    }
+
+    /// Utils
```

Comment:
would probably be nice to re-use the helper methods we have on `AATestScaffolding.sol`, but not big deal, we can leave it as is and then I can refactor it

Reply:
Yeah, my thinking as well. They are slighted different than methods at  `AATestScaffolding`. I just didn't want to modify `AATestScaffolding` for now.